n3fit/src/n3fit/msr.py

```diff
@@ -86,13 +86,17 @@ def generate_msr_model_and_grid(
     pdf_integrated = xIntegrator(weights_array, input_shape=(nx,))(pdf_integrand)
 
     # 5. THe input for the photon integral, will be set to 0 if no photons
-    photon_integral = Input(shape=(), batch_size=1, name='photon_integral')
+    photon_integral = Input(shape=(1,), batch_size=1, name='photon_integral')
+
+    # 5b. Insert the photon integral as the first component of the pdf integrals
+    pdf_integrated = Lambda(
+        lambda pdf_photon: op.concatenate([pdf_photon[1], pdf_photon[0][:, 1:]], axis=1),
+        name="join_photon",
+    )([pdf_integrated, photon_integral])
 
     # 6. Compute the normalization factor
     # For now set the photon component to None
-    normalization_factor = MSR_Normalization(output_dim, mode, name="msr_weights")(
-        pdf_integrated, photon_integral
-    )
+    normalization_factor = MSR_Normalization(mode, name="msr_weights")(pdf_integrated)
```

Comment:
Why this change? I feel the previous one was actually much clearer since it makes the difference between the PDF integration and the photon explicit in the MSR.

(and they are really different objects, while the PDF comes from the NN, the photon comes from an analytical computation and is constant throughout the fit, so the separation in the `call` is good)

Reply:
Maybe I misunderstand (I never understood why the photon can't be treated the same) but they're not really different objects right? They might arise differently but they are all integrals right? I thought a dedicated layer that inserts the photon integral in the first component makes it more transparent, and simplifies the MSR layer.
But if you still prefer the old way I'll change it back.

Reply:
The photon is analytically computed before the fit 
https://github.com/NNPDF/nnpdf/blob/9cf5f9ecc83ec59fff804f859b531198139021fe/validphys2/src/validphys/photon/compute.py#L125C13-L125C13

Computing the integral in the same way of the others would require too many points and the photon contribution is not large so we don't need the same level of precision.

So, while they are all "the result of integrals", the photon is a constant that gets injected into the calculation while the others are quantities that are computed and change from epoch to epoch.

Reply:
Ok, I changed it back to what it was.

Any idea why the CI is failing? Thought it was some fluke but it happend twice in a row now, just says the operation was cancelled when solving the environment..